src/nnsight/contexts/session/Session.py

```diff
@@ -55,7 +56,10 @@ def iter(self, iterable, **kwargs) -> Iterator:
 
     def local_backend_execute(self) -> Dict[int, Graph]:
 
-        super().local_backend_execute()
+        try:
```

Comment:
Why do you need to check for the Exception in all the subclasses if you are doing it in GraphBasedContext?

Reply:
Because in the case of the `Iterator`, we want to break out of the loop, which is done upon catching the exception.